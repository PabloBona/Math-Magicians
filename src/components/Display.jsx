import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ calculatorData }) => (
  <div className="col-8 col-6 d-flex justify-content-center">
    <div className="d-flex justify-content-end text-light px-1 py-2 w-100 mx-5">
      {calculatorData ? (calculatorData.next || calculatorData.total || '0') : '0'}
    </div>
  </div>
);

Display.propTypes = {
  calculatorData: PropTypes.shape({
    total: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    next: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    operation: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  }),
};

Display.defaultProps = {
  calculatorData: null,
};

export default Display;
