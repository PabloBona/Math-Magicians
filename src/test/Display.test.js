import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from '../components/Display';

describe('Display component unit test', () => {
  test('Display Component Renders Properly', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('The Display Component Displays Result', () => {
    render(<Display calculatorData={{ total: '3.14' }} />);
    expect(screen.getByText('3.14', { exact: false })).toBeInTheDocument();
  });
});
