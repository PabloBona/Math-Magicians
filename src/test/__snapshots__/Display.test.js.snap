// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component unit test Display Component Renders Properly 1`] = `
<div
  className="col-8 col-6 d-flex justify-content-center"
>
  <div
    className="d-flex justify-content-end text-light px-1 py-2 w-100 mx-5"
  >
    0
  </div>
</div>
`;
